pkg/kube/vault/vault.go

```diff
@@ -121,19 +144,90 @@ func SystemVaultNameForCluster(clusterName string) string {
 	return cluster.ShortNameN(fullName, 22)
 }
 
-// CreateVault creates a new vault backed by GCP KMS and storage
-func CreateVault(kubeClient kubernetes.Interface, vaultOperatorClient versioned.Interface, name string, ns string,
+// CreateGKEVault creates a new vault backed by GCP KMS and storage
+func CreateGKEVault(kubeClient kubernetes.Interface, vaultOperatorClient versioned.Interface, name string, ns string,
 	gcpServiceAccountSecretName string, gcpConfig *GCPConfig, authServiceAccount string,
 	authServiceAccountNamespace string, secretsPathPrefix string) error {
 
+	vault, err := InitializeVault(kubeClient, name, ns, authServiceAccount, authServiceAccountNamespace, secretsPathPrefix)
+	if err != nil {
+		return err
```

Comment:
wrap err

Reply:
Is it really needed when all errors returned from InitializeVault are wrapped?